packages/forms/backend/sirius-components-collaborative-widget-reference/src/main/java/org/eclipse/sirius/components/collaborative/widget/reference/ReferenceWidgetDescriptionConverterSwitch.java

```diff
@@ -353,6 +353,14 @@ private IStatus handleAddReference(VariableManager variableManager, ReferenceWid
 
     private Object handleCreateElement(VariableManager variableManager, ReferenceWidgetDescription referenceDescription) {
         Optional<Object> result = Optional.empty();
+        if (this.isContainment(variableManager, referenceDescription) && !this.isMany(variableManager, referenceDescription)) {
+            // If the reference is mono valued and containment, we must first unset value before create a new one
+            EObject owner = this.getReferenceOwner(variableManager, referenceDescription.getReferenceOwnerExpression());
+            String referenceName = this.getStringValueProvider(referenceDescription.getReferenceNameExpression()).apply(variableManager);
+            if (owner != null && owner.eClass().getEStructuralFeature(referenceName) instanceof EReference reference) {
+                owner.eUnset(reference);
+            }
+        }
```

Comment:
I'd put this inside the if (optionalIsChild.isPresent() && optionalEditingContext.isPresent()) block below, to avoid modifying the reference in cases where we discover a few lines later that we can not actually create the new element.

Reply:
You're right, I've moved it